wondrous-app/components/Common/TaskMenuStatus.tsx

```diff
@@ -0,0 +1,287 @@
+} from 'utils/constants';
+import { parseUserPermissionContext } from 'utils/helpers';
+import { useOrgBoard, usePodBoard, useUserBoard } from 'utils/hooks';
+
+const TaskStatusMenuWrapper = styled(Menu)`
+  && {
+    .MuiMenu-list,
+    .MuiMenu-paper {
+      padding: 0;
+      background-color: ${({ theme }) => theme.palette.midnight};
+      min-width: 150px;
+      outline: 1px solid ${({ theme }) => theme.palette.grey79};
+    }
+  }
+`;
+
+const TaskStatusMenuItem = styled(MenuItem)`
+  && {
+    background-color: ${({ theme }) => theme.palette.midnight};
+    display: flex;
+    align-items: center;
+    gap: 8px;
+    height: 28px;
+    padding: 0 6px;
+    :hover {
+      background: ${({ theme }) => theme.palette.grey90};
+    }
+    > svg {
+      width: 18px !important;
+      height: 18px !important;
+    }
+  }
+`;
+
+const TaskStatusMenuButtonArrow = styled((props) => (
+  <div {...props}>
+    <Arrow />
+  </div>
+))`
+  && {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 9px;
+    transform: rotate(${({ open }) => (open ? `-90` : `90`)}deg);
+    svg {
+      path {
+        fill: white;
+      }
+    }
+  }
+`;
+
+export const TaskStatusMenuButton = styled(ButtonBase)`
+  && {
+    outline: ${({ open, theme }) => open && `1px solid ${theme.palette.grey79}`};
+    background-color: ${({ theme }) => theme.palette.midnight};
+    min-width: fit-content;
+    border-radius: 6px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 0 6px;
+    gap: 8px;
+    height: 28px;
+    && {
+      > svg {
+        width: 18px !important;
+        height: 18px !important;
+      }
+    }
+  }
+`;
+
+const TaskModalStatusLabel = styled(Typography)`
+  && {
+    font-size: 14px;
+    line-height: 0;
+    ${({ theme }) => `
+    font-weight: ${theme.typography.fontWeightRegular};
+    color: ${theme.palette.white};
+  `}
+  }
+`;
+
+const refetchNonProposalQueries = [
+  GET_ORG_TASK_BOARD_TASKS,
+  GET_PER_STATUS_TASK_COUNT_FOR_ORG_BOARD,
+  GET_PER_STATUS_TASK_COUNT_FOR_POD_BOARD,
+  GET_PER_STATUS_TASK_COUNT_FOR_USER_BOARD,
+  GET_POD_TASK_BOARD_TASKS,
+  GET_TASK_BY_ID,
+  GET_TASKS_FOR_MILESTONE,
+  GET_USER_TASK_BOARD_TASKS,
+];
+
+const refetchTaskProposalQueries = [
+  GET_ORG_TASK_BOARD_PROPOSALS,
+  GET_POD_TASK_BOARD_PROPOSALS,
+  GET_TASK_PROPOSAL_BY_ID,
+  GET_USER_TASK_BOARD_PROPOSALS,
+];
+const taskProposalStatus = (task) => {
+  if (task?.approvedAt) return STATUS_APPROVED;
+  if (task?.closedAt) return STATUS_CLOSED;
+  return STATUS_OPEN;
+};
+
+const statusesProposal = ({ task, entityType }) => {
+  const filterStatus = ENTITIES_TYPES_FILTER_STATUSES({ orgId: task?.orgId })[entityType]?.filters[0].items;
+  const status = taskProposalStatus(task);
+  const currentStatus = filterStatus?.find((i) => i.id === status);
+  return { filterStatus, currentStatus };
+};
+
+const statusesNonProposal = ({ task, entityType, canArchive }) => {
```

Comment:
think we need a more explanatory name here, how about getProposalStatuses and getTaskStatuses ?